src/TokenFuzzChecks.sol

```diff
@@ -0,0 +1,461 @@
+            assertEq(TokenLike(_token).balanceOf(to), prevToBalance + amount);
+        }
+    }
+
+    function fuzzCheckTransferFrom(
+        address _token,
+        address to,
+        uint256 approval,
+        uint256 amount
+    ) public {
+        vm.assume(to != address(0) && to != _token);
+        uint256 prevToBalance = TokenLike(_token).balanceOf(to);
+        approval = bound(approval, 0, type(uint256).max - prevToBalance);
+        amount = bound(amount, 0, approval);
+        address from = address(0xABCD);
+        deal(_token, from, TokenLike(_token).balanceOf(from) + amount, true);
+        uint256 prevSupply = TokenLike(_token).totalSupply();
+        uint256 prevFromBalance = TokenLike(_token).balanceOf(from);
+        vm.prank(from);
+        TokenLike(_token).approve(address(this), approval);
+
+        vm.expectEmit(true, true, true, true);
+        emit Transfer(from, to, amount);
+        assertTrue(TokenLike(_token).transferFrom(from, to, amount));
+
+        assertEq(TokenLike(_token).totalSupply(), prevSupply);
+        uint256 app = from == address(this) || approval == type(uint256).max ? approval : approval - amount;
+        assertEq(TokenLike(_token).allowance(from, address(this)), app);
+        if (from == to) {
+            assertEq(TokenLike(_token).balanceOf(from), prevFromBalance);
+        } else  {
+            assertEq(TokenLike(_token).balanceOf(from), prevFromBalance - amount);
+            assertEq(TokenLike(_token).balanceOf(to), prevToBalance + amount);
+        }
+    }
+
+    function fuzzCheckTransferInsufficientBalance(
+        address _token,
+        string memory _contractName,
+        address to,
+        uint256 mintAmount,
+        uint256 sendAmount
+    ) public {
+        vm.assume(to != address(0) && to != address(_token));
+        uint256 prevSupply = TokenLike(_token).totalSupply();
+        mintAmount = bound(mintAmount, 0, type(uint256).max - prevSupply - 1);
+        sendAmount = bound(sendAmount, mintAmount + 1, type(uint256).max);
+        deal(_token, address(this), TokenLike(_token).balanceOf(address(this)) + mintAmount, true);
+
+        vm.expectRevert(abi.encodePacked(_contractName, "/insufficient-balance"));
+        TokenLike(_token).transfer(to, sendAmount);
+    }
+
+    function fuzzCheckTransferFromInsufficientBalance(
+        address _token,
+        string memory _contractName,
+        address to,
+        uint256 mintAmount,
+        uint256 sendAmount
+    ) public {
+        vm.assume(to != address(0) && to != address(_token));
+        uint256 prevSupply = TokenLike(_token).totalSupply();
+        mintAmount = bound(mintAmount, 0, type(uint256).max - prevSupply - 1);
+        sendAmount = bound(sendAmount, mintAmount + 1, type(uint256).max);
+        address from = address(0xABCD);
+        deal(_token, from, TokenLike(_token).balanceOf(from) + mintAmount, true);
```

Comment:
Same here.

Reply:
Good catch, changed.

Reply:
Hmmm instead of `fuzzCheckTransferFromInsufficientBalance` it was changed the `deal` function of `fuzzCheckTransferFromInsufficientAllowance ` which actually could stay minting over the prior balance as it is only checking the allowance part reverts.

Reply:
Yes, I've done another pass in TokenFuzzChecks to improve the robustness of these tests against edge cases so I think now they should all be correct. Using deal() without taking into account the prior balance was sometimes causing some overflow with the totalSupply further down the line so now our usage of deal() is always taking into account the previous balance, which is also easier to reason about imo.